### DISCUSSION TITLE
13. June 2023

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#1385] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [x] Li Xun: Review design of EDMM: Add support for dynamic thread creation

#### Li Xun: Review design of EDMM: Add support for dynamic thread creation

See https://github.com/gramineproject/gramine/issues/1223#issuecomment-1521299680.

[ Li Xun explains the diagram ]

Dmitrii asked many technical questions. He thinks that the design is sound. Others didn't participate in the discussion, so looks like this design is not opposed.

Note that Intel SGX SDK outsources all this complexity to the application developer. In Gramine, the LibOS + PAL need to do all the flows themselves.

Li Xun already has a PoC.

< Unfortunately, I was engrossed in the discussion and couldn't keep the notes >

**Comment by llly** [comment]
Review design of `EDMM: Add support for dynamic thread creation`https://github.com/gramineproject/gramine/issues/1223#issuecomment-1521299680
